### PR TITLE
WSL: avoid error when listing no distributions.

### LINKED ISSUE #1655: RD does not start if no WSL distros exist
### Actual Behavior

RD does not start, with no UI after the first run dialog.

### Steps to Reproduce

- Start with a new VM (or delete all your WSL distributions)
- Try to start RD

### Result

`background.log`:
```
2022-02-24T18:54:33.836Z: Error starting up: Error: wsl.exe exited with code 4294967295
    at ChildProcess.<anonymous> (C:\Users\Joe\AppData\Local\Programs\Rancher Desktop\resources\app.asar\dist\app\background.js:1:8644)
    at ChildProcess.emit (node:events:394:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12) {
  stdout: 'Windows Subsystem for Linux has no installed distributions.\r\r\n' +
    'Distributions can be installed by visiting the Microsoft Store:\r\r\n' +
    'https://aka.ms/wslstore\r\r\n'
}
```

`wsl.log`:
```
2022-02-24T18:54:33.513Z: Running command wsl --list --verbose...
2022-02-24T18:54:33.834Z: WSL failed to execute wsl.exe --list --verbose: Error: wsl.exe exited with code 4294967295
```

### Expected Behavior

RD starts correctly.

### Additional Information

Fallout from #1637 — `wsl --list --quiet` returns success (and no results) if there's nothing. `wsl --list --verbose` returns `-1` in the same case.

### Rancher Desktop Version

1.0.0-343-ga2eb3bb

### Rancher Desktop K8s Version

N/A

### Which container runtime are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 11 Home 21H2 (Build 22000.194)

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

_No response_

### Windows User Only

Running in a VM, no other networking.

### PR DESCRIPTION
**Pull request by mook-as**
`wsl --list --verbose` fails if there is no distributions to list; this is unlike `wsl --list --quiet` which succeeds with no entries.

Fixes #1655

### CODE REVIEW

File: src/k8s-engine/wsl.ts [hunk 1]
```diff
@@ -412,16 +424,7 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
...
-      const runningOut = await this.execWSL({ capture: true }, '--list', '--quiet', '--running');
-      const running = runningOut.split(/[\r\n]+/).map(x => x.trim()).filter(x => x);
-
-      result = result.filter(x => running.includes(x));
-    }
-
-    console.debug(`${ runningOnly ? 'Running' : 'All' } distros: ${ result }`);
-
-    return result;
+    return result.filter(x => distros.includes(x));
```
Comment (ericpromislow) [inline src/k8s-engine/wsl.ts:427]: What about `return !runningOnly ? result : result.filter(x => distros.includes(x))` ?
More code, but no need to intersect the lists if we want all of them.

**Review by ericpromislow — APPROVED** [review]
Code's ok and works.

File: src/k8s-engine/wsl.ts (hunk 1 above)
Comment (mook-as) [reply src/k8s-engine/wsl.ts:427]: Makes sense; my tree is busy, but I think I'll do `return result.filter(x => runningOnly || distros.includes(x));` (relying on short circuiting to bypass the extra evaluation).

Comment (jandubois) [reply src/k8s-engine/wsl.ts:427]: I don't know; this logic is kind of hard to understand, and you are optimizing a filter of less than 20 items.

Given that we don't have tests for the logic, I would prefer the simpler version. We already spend more time "talking" about this than this optimization will ever save us.

Comment (mook-as) [reply src/k8s-engine/wsl.ts:427]: Alright, given the short expected list, I'll keep things as-is; we can revisit if people report having a bazillion WSL distros.

**Pull request merged**